src/sys/DISK.ASM

```diff
@@ -958,6 +938,228 @@ disk_op:                                        ; Perform a disk operation using
+    clc                                         ; No error
+
+.done:
+    mov dx, word [es:si]                        ; Read the corresponding entry in the FAT (or what we just wrote)
+    mov word [register_bckp_a], dx              ; Preserve entry
+
+    popa                                        ; Restore register states
+    mov dx, word [register_bckp_a]              ; Restore entry
+
+    ret                                         ; Return to caller
+
+.invalid:
+    mov ax, 0x0308                              ; Invalid cluster number
+    call error                                  ; Handle this error
+
+    jmp .done                                   ; We're done
+
+.abort:
+    popf                                        ; Restore for balance
+
+    call abort                                  ; Aborting message
+    stc                                         ; Carry-flag for error
+
+    jmp .done                                   ; We're done
+
+
+; INTERPRET FAT ENTRY: Retrieves the next entry from the FAT and sets flags accordingly
+; IN: [cluster] = Current cluster
+; OUT: [cluster] = Next cluster; ZF = Set for EOF; CF = Set on error
+
+interpret_fat:                                  ; Interpret entry from FAT
+    pusha                                       ; Save register states to stack
+
+    clc                                         ; Retrieve entry
+    call fat_entry                              ; Get next entry from FAT
+    jc .done                                    ; If error, we're done
+
+.handle:
+    cmp dx, 0xFFF8                              ; Is this the end of the file?
+    jae .end_file                               ; If so, return appropriately
+
+    cmp dx, 0xFFF7                              ; Is this marked as bad?
+    je .error_bsec                              ; If so, error
+
+    cmp dx, 0x0002                              ; Is this an invalid cluster number?
+    jbe .error_fat                              ; If so, error
+
+    cmp dx, word [gs:cluster]                   ; Is this the cluster we just loaded?
+    je .error_fat                               ; If so, error
+
+    mov word [gs:cluster], dx                   ; Otherwise, this is the next cluster
+
+.done:
+    popa                                        ; Restore register states
+    ret                                         ; Return to caller
+
+.end_file:
+    xor ax, ax                                  ; Set zero-flag
+    jmp .done                                   ; We're done
+
+.error_fat:
+    mov ax, 0x0305                              ; Inconsistent FAT
+    call error                                  ; Handle this error
+
+    jmp .done                                   ; We're done
+
+.error_bsec:
+    mov ax, 0x0304                              ; Cluster marked bad
+    call error                                  ; Handle this error
+
+    jmp .done                                   ; We're done
+
```

Comment:
Created separate routines for retrieving entries in the FAT in order to simplify the process for implementing a dynamically loaded FAT, as well as FAT12 (coming soon).